app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharecterView.kt

```diff
@@ -50,4 +67,44 @@ class CharecterView(activity: MainActivity) {
         activityRef.get()!!.progressBar.visibility = View.VISIBLE
 
     }
+
+
+
+    fun showCharacterInformation(character: Character) {
+        val dialogFragment = CharacterDialogFragment()
+        val args = Bundle()
+        args.putParcelable(EXTRA_CHARACTER, character)
+        dialogFragment.arguments = args
+        dialogFragment.show(activityRef.get()?.supportFragmentManager, "txn_tag")
+    }
+
+    class CharacterDialogFragment() : DialogFragment() {
+
+        private lateinit var character: Character
+
+
+        override fun onCreate(savedInstanceState: Bundle?) {
+            super.onCreate(savedInstanceState)
+        }
+
+        override fun onStart() {
+            super.onStart()
+            val d: Dialog? = dialog
+            if (d != null) {
```

Comment:
please in kotlin try to not check null. You can use something like Dialog?.let { 
val ...
val ...
it.window?.setLayout(...)
}

Reply:
Done